### ISSUE TITLE
MessageHandlerException reason not available.

### ISSUE THREAD

**Issue by kedargh** [microsoft/autogen#5589]
I am trying to run a MagenticOneGroupChat with 4 agents in play. 

```python
market_research_agent = AssistantAgent(
    name="MarketResearchAgent",
    description="Fetches the financial indicators and other attributes of a stock.",
    model_client=model_client,
    tools=[fetch_company_details],
    system_message="You are a helpful assistant. Use tools when needed."
)

news_agent = AssistantAgent(
    name="NewsFetcherAgent",
    description="Fetches financial news for a given stock ticker.",
    model_client=model_client,
    tools=[headlines_fetcher],
    system_message="You are a helpful assistant. Use tools when needed.",

)
sentiment_classifier_agent = AssistantAgent(
    name="SentimentClassifierAgent",
    description="Classifies sentiment for a list of news articles. Used for sentiment analysis. Use the sentiment_classifier_loop for classifying news as positive, negative or neutral.",
    model_client=model_client,
    tools=[sentiment_classifier_loop], #headlines_fetcher , , sentiment_classifier_loop
    system_message="You are a helpful assistant. Use tools when needed.",
)

user_agent = UserProxyAgent(
    name="AskerAgent",
    description="A human user",
    input_func=input,
    )


```

The prompt given to the groupchat is as follows - 

```python
task_to_perform = """Perform sentiment analysis for the given stock. The report should be structured in Markdown format and include the following sections:  

## 1. Recent News & Market Sentiment  
- Fetch **top 10 recent headlines** related to the stock.  
- Classify each headline as **Positive, Negative, or Neutral** using sentiment analysis.  

---
### **Stock to Analyze:**  
**Company Name**: Infosys  
**Ticker**: INFY  

Use the given agents to you for doing all the work. The final report should be **structured in Markdown format** for easy readability.
 """

    stream = team.run_stream(task=task_to_perform)
```

It is doing the first task that is fetching the news. But as soon as it tries to call the "sentiment_classifer_agent" it gives the following exception - 

"],\"type\":\"ToolCallExecutionEvent\"}]}}", "handling_agent": "group_chat_manager/b3cd3d6d-897a-47e5-8d4d-4f41f745a93b", "exception": "", "type": "MessageHandlerException"}"

How can I make this work? I apologize for the format of this issue to the person contributor viewing this, this is my first time asking a question :)

**Comment by jackgerrits** [comment]
Can you please provide the full code/script that repros the issue? It's hard to diagnose without it.

**Comment by kedargh** [comment]
Yes sure. Sorry for the late reply. I guess it also has to do with OpenAI not allowing good amount of chain of thoughts support for Gpt-4o-mini. Here's the code - 

"""
from gnews import GNews
import csv
from autogen_agentchat.agents import AssistantAgent
from autogen_agentchat.base import TaskResult
from autogen_agentchat.conditions import ExternalTermination, TextMentionTermination
from autogen_agentchat.teams import RoundRobinGroupChat
from autogen_agentchat.agents import UserProxyAgent
from pydantic import BaseModel
from typing import List, Dict
from autogen_agentchat.ui import Console
from autogen_agentchat.messages import TextMessage
from autogen_agentchat.ui import Console
from autogen_core import CancellationToken
from autogen_ext.models.openai import OpenAIChatCompletionClient
import openai
from transformers import AutoModelForSequenceClassification, AutoTokenizer
import torch
from news_fetcher import fetch_news_for_ticker
import asyncio
from rich.console import Console
import yfinance as yf
from autogen_agentchat.teams import MagenticOneGroupChat
#################MESSAGE FORMATS FOR AGENTS#############################################
class NewsResponse(BaseModel):
    stock: str
    headlines: List[str]

class SentimentResponse(BaseModel):
    stock: str
    results: List[Dict[str, str]]

class StockInfo(BaseModel):  #####PYDANTIC FOR STOCK ATTRIBUTES######
    stock: str
    attributes: Dict



############################################################################################################################
model_client = OpenAIChatCompletionClient(
    model="gpt-4o-mini",
    api_key=<put-your-4o-mini-key-here, Mr. Issue Resolver>,
)
################################################################################################################################
async def headlines_fetcher(stock : str , number_of_news: int)->NewsResponse:         #####TOOL 1####
    """Fetch the list of news for the given stock ticker"""
    google_news = GNews(
    language='en',
    country='US',
    period='1d',
    start_date=None,
    end_date=None,
    max_results=number_of_news,)
    ####GOOGLE NEWS OBJECT#####
    news = google_news.get_news(f"{stock}")
    headlines = []
    for single_news in news:
        headlines.append(single_news['title'])

    for headline in headlines:
        headline = headline.split('-')[0].strip()
    
    #return headlines
    return NewsResponse(stock=stock, headlines=headlines)


model_path = "yiyanghkust/finbert-tone"
tokenizer = AutoTokenizer.from_pretrained(model_path)
model = AutoModelForSequenceClassification.from_pretrained(model_path)
model.eval()


async def sentiment_classifier_loop(news_response_object : NewsResponse) -> SentimentResponse:
    """Classify sentiment for multiple news articles asynchronously and return structured response."""
    if isinstance(news_response_object, dict):
        news_response_object = NewsResponse(**news_response_object)

    headlines = news_response_object.headlines

    async def sentiment_classifier(news: str) -> Dict[str,str]:
        """Single news classification"""
        inputs = tokenizer([news], return_tensors="pt", truncation=True, padding=True)

        with torch.no_grad():
            outputs = model(**inputs)
            predictions = torch.softmax(outputs.logits, dim=1)
            label_index = torch.argmax(predictions).item()
        labels = ["Neutral", "Positive", "Negative"]
        return {news: labels[label_index]}

    tasks = [sentiment_classifier(news) for news in headlines]
    results = await asyncio.gather(*tasks)

    return SentimentResponse(stock=news_response_object.stock , results=results)


async def fetch_company_details(stock_name: str)-> StockInfo:
    data = yf.Ticker(stock_name)
    dict_for_analyst_price_targets = data.analyst_price_targets
    dict_for_ticker_info = data.info
    dict_for_analyst_price_targets.update(dict_for_ticker_info)

    return StockInfo(stock=stock_name,attributes=dict_for_analyst_price_targets)


#####################AGENTS###################################################################
market_research_agent = AssistantAgent(
    name="MarketResearchAgent",
    description="Fetches the numerical characteristics and other attributes of a stock.",
    model_client=model_client,
    tools=[fetch_company_details],
    system_message="You are a helpful assistant. Use tools when needed."
)
###IS IT RIGHT TO HAVE AN AGENT FOR NEWS SEPARATELY OR SHOULD IT BE GIVEN THE FUNCTIONTOOLS TO PERFORM NUMERICAL DATA RESEARCH???
news_agent = AssistantAgent(
    name="NewsFetcherAgent",
    description="Fetches financial news for a given stock ticker.",
    model_client=model_client,
    tools=[headlines_fetcher],
    system_message="You are a helpful assistant. Use tools when needed.",

)
sentiment_classifier_agent = AssistantAgent(
    name="SentimentClassifierAgent",
    description="Classifies sentiment for a list of news articles. Used for sentiment analysis.",
    model_client=model_client,
    tools=[sentiment_classifier_loop], #headlines_fetcher , , sentiment_classifier_loop
    system_message="You are a helpful assistant. Use tools when needed.",
)

user_agent = UserProxyAgent(
    name="AskerAgent",
    description="A human user",
    input_func=input,
    )
################################################################################################################
# async def main():

team = MagenticOneGroupChat([user_agent, market_research_agent , news_agent , sentiment_classifier_agent], model_client=model_client) #, termination_condition="DONE")

#stream=team.run_stream(task="""Perform sentiment analysis for Tata Consultancy Services.""")


async def run_chat():
    task_to_perform = str(input("Enter the task to perform    :    "))

    stream = team.run_stream(task=task_to_perform)

    console = Console()
    async for result in stream:
        console.print(result)

asyncio.run(run_chat())

"""


Put your Gpt-4o-mini key in the tag given. Let me know your insights or a particular way I should structure my prompts properly for the LLM to understand.
